File: library-core/src/main/java/com/mikepenz/fastadapter/ISwipeable.java
package com.mikepenz.fastadapter;

/**
 * Created by mikepenz on 30.12.15.
 */
public interface ISwipeable<T, Item extends IItem> {
    /**
     * @return true if swipeable
     */
    boolean isSwipeable();

    /**
     * use this method to set if item is swipeable
     *
     * @param swipeable true if swipeable
     * @return this
     */
    T withIsSwipeable(boolean swipeable);
}

File: fastadapter-extensions-swipe/src/main/java/com/mikepenz/fastadapter/swipe/ISwipeable.kt
package com.mikepenz.fastadapter.swipe

import androidx.recyclerview.widget.ItemTouchHelper

/**
 * Created by mikepenz on 30.12.15.
 */
interface ISwipeable {
    /** @return true if swipeable */
    val isSwipeable: Boolean

    /**
     * @return true if the provided direction is supported
     *
     * This function will only be called if [isSwipeable] returns true
     */
    fun isDirectionSupported(direction: Int) : Boolean = true
}
